### PR TITLE
Datadok trying many possible paths + utility functions

### PR DESCRIPTION
**Pull request by aecorn**
Bumps version to 1.0.2

Main functionality added is 
- a broader search in datadok_extract to look for valid paths in the API and on disk. We are even trying to look BACK IN TIME to find possible matching entries in Datadok...
- Function in saspy for easily storing dataframes as sas7bdat-files
- Function in statistikkregisteret to raise an error on a missing future publishing (when we start work on a statistical product, the publishing should already have been reported in 3 months before the publishing date).

Other points:
- Removed duplicate type hints for Args in the docstrings.
- Trying to work on suggestions from Sonarcloud.

image
image

### CODE REVIEW

File: tests/data/test_datadoc_extract.py [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+        "datatype": ["Desimaltall", "Desimaltall"],
+        "precision": [None, None],  # Precision not used in this case
+    }
+    metadata_df = pd.DataFrame(metadata_data)
+
+    df, metadata_df = datadok_extract.handle_decimals(df, metadata_df)
+
+    assert df["col1"].dtype == "Float64"
+    assert df["col2"].dtype == "Float64"
+    assert df["col1"].iloc[0] == 1.23
```
Comment (arneso-ssb) [inline tests/data/test_datadoc_extract.py:216]: Equality checks with floating points values are unsafe, see https://rules.sonarsource.com/python/RSPEC-1244/
Use `math.isclose` to solve this, like:
```python
assert math.isclose(df["col1"].iloc[0], 1.23)
```

**Review by arneso-ssb — DISMISSED** [review]
Looks good, just two minor suggestions.

File: tests/data/test_datadoc_extract.py (hunk 1 above)
Comment (aecorn) [reply tests/data/test_datadoc_extract.py]: Nice thanks, will do

File: src/fagfunksjoner/api/statistikkregisteret.py [hunk 2]
```diff
@@ -588,3 +616,62 @@ def etree_to_dict(t: ET.Element) -> dict[str, Any]:
...
+        msg = f"Cant find any publishing times for {shortname}. Are you using the correct shortname?"
+        if raise_error:
+            raise FuturePublishingError(msg)
+        else:
+            logger.warning(msg)
+            return next_time
+
+    elif isinstance(next_time, datetime.timedelta) and next_time < zerotime:
+        msg = f"You haven't added a next publishing to the register yet? {next_time.days} days since last publishing?"
+        if raise_error:
```
[the remark this answers is not in the document]
Comment (aecorn) [reply src/fagfunksjoner/api/statistikkregisteret.py]: Good idea, changed it slightly to not pass next_time through.

**Review by arneso-ssb — APPROVED** [review]

**Pull request merged**